pkg/prometheus/promcfg.go

```diff
@@ -868,11 +868,14 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 	cfg = cg.AddTrackTimestampsStaleness(cfg, ep.TrackTimestampsStaleness)
 
 	var attachMetaConfig *attachMetadataConfig
-	if m.Spec.AttachMetadata != nil {
+	if m.Spec.AttachMetadata != nil || scrapeClass.AttachMetadata != nil {
 		attachMetaConfig = &attachMetadataConfig{
 			MinimumVersion: "2.35.0",
 			AttachMetadata: m.Spec.AttachMetadata,
 		}
+		if m.Spec.AttachMetadata == nil {
```

Comment:
The configs defined on the monitor objects (PodMonitor, ServiceMonitor) are always taking precedence over scrape class.

It means the m.Spec.AttachMetadata will always have precedence over the scrapeClass.AttachMetadata.

Reply:
Maybe worth create a function named GetAttachedMetadata and we implement this precedency logic inside this function and just reuse it, wdyt?

Reply:
> The configs defined on the monitor objects (PodMonitor, ServiceMonitor) are always taking precedence over scrape class.
> 
> It means the m.Spec.AttachMetadata will always have precedence over the scrapeClass.AttachMetadata.

Yes, I thought that was the intention for Scrape classes
 As per the [proposal](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202305-scrapeclasses.md#non-goals) its a non-goal to `Provide a way for Prometheus administrators to enforce/override settings on scrape configurations.`

> Maybe worth create a function named GetAttachedMetadata and we implement this precedency logic inside this function and just reuse it, wdyt?

Yeah that's probably cleaner, will do.